src/diagnostics/core_diagnostics.jl

```diff
@@ -733,3 +733,315 @@ add_diagnostic_variable!(
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "prw",
+    long_name = "Water Vapor Path",
+    standard_name = "atmospheric_mass_content_of_water_vapor",
+    units = "kg m^-2",
+    comments = "Vertically integrated specific humidity",
+    compute! = compute_prw!,
+)
+
+###
+# Total Column Integrated Hurvi (1d)
+###
+compute_hurvi!(out, state, cache, time) =
+    compute_hurvi!(out, state, cache, time, cache.atmos.moisture_model)
+compute_hurvi!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("hurvi", model)
+
+function compute_hurvi!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        out = zeros(axes(Fields.level(state.f, half)))
+        saturation = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            saturation,
+            LB.@lazy_broadcasted @. (
+                state.c.ρ *
+                TD.q_vap_saturation(thermo_params, cache.precomputed.ᶜts)
+            )
+        )
+        Operators.column_integral_definite!(
+            out,
+            LB.@lazy_broadcasted @. (state.c.ρq_tot / saturation)
+        )
+        return out
+    else
+        saturation = zeros(axes(Fields.level(state.f, half)))
+        Operators.column_integral_definite!(
+            saturation,
+            LB.@lazy_broadcasted @. (
+                state.c.ρ *
+                TD.q_vap_saturation(thermo_params, cache.precomputed.ᶜts)
+            )
+        )
+        Operators.column_integral_definite!(
+            out,
+            LB.@lazy_broadcasted @. (state.c.ρq_tot / saturation)
+        )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "hurvi",
+    long_name = "Relative Humidity Saturation-Weighted Vertical Integral",
+    standard_name = "relative_humidity_vi",
+    units = "kg m^-2",
+    comments = "Integrated relative humidity over the vertical column",
+    compute! = compute_hurvi!,
+)
+
+
+###
+# Vapor specific humidity (3d)
+###
+compute_vhus!(out, state, cache, time) =
+    compute_vhus!(out, state, cache, time, cache.atmos.moisture_model)
+compute_vhus!(_, _, _, _, model::T) where {T} =
+    error_diagnostic_variable("vhus", model)
+
+function compute_vhus!(
+    out,
+    state,
+    cache,
+    time,
+    moisture_model::T,
+) where {T <: Union{EquilMoistModel, NonEquilMoistModel}}
+    thermo_params = CAP.thermodynamics_params(cache.params)
+    if isnothing(out)
+        return TD.vapor_specific_humidity.(
+            CAP.thermodynamics_params(cache.params),
+            cache.precomputed.ᶜts,
+        )
+    else
+        out .=
+            TD.vapor_specific_humidity.(
+                CAP.thermodynamics_params(cache.params),
+                cache.precomputed.ᶜts,
+            )
+    end
+end
+
+add_diagnostic_variable!(
+    short_name = "vhus",
```

Comment:
```suggestion
    short_name = "husv",
```
(I think technically this is `hus`, but we can change it later.)